### DISCUSSION TITLE
Common Services with Epic Stack

### DISCUSSION THREAD

**Discussion by mhdrad** [epicweb-dev/epic-stack#532] [Ideas]
Many software projects and SaaS require common functionality like content management, and billing and subscription management. Although open-source libraries exist for these, but whenever you need changes, it's become a real headache as they are not in your tech stack and make it hard to maintain.

I propose creating an open-source repository using epic epic-stack that provides reusable modular services. Therefore you can clone into your codebase (just like Shadcn) and you can easily apply changes and keep it updated with the repo:

Potential services:
- Role-based authorization
- CMS with dynamic fields
- Subscription billing management

This would save development time and prevent duplicate work across projects. The open-source community could collaborate to enhance and maintain the services over time.

Please share your thoughts on creating standardized open-source modules for common needs. I believe this could benefit many development teams and products.

**Comment by kentcdodds** [comment]
This sounds like an awesome idea. I'm afraid I don't have the bandwidth to work on it, but I think if someone could it would help a lot of people!

I'm going to close this, not because I don't think it's a good idea, but because I definitely won't have time to work on it. Sorry!
